wherehows-web/app/components/datasets/health/score-table.ts

```diff
@@ -0,0 +1,182 @@
+      class: `dataset-health__score-table__${header.toLowerCase()}`,
+      dropdownOptions: dropdownOptions[header.toLowerCase()],
+      initialDropdown: { label: header, value: '' },
+      onDropdownSelect: onDropdownSelect.bind(this, header)
+    }));
+  });
+
+  /**
+   * Uses the passed in table data to calculate a dropdown list of options available for filtering by category
+   * and severity for the table's rows.
+   * @type {ComputedProperty<IHealthTableDropdowns>}
+   */
+  dropdownOptions: ComputedProperty<IHealthTableDropdowns> = computed('tableData', function(
+    this: DatasetsHealthScoreTable
+  ) {
+    const tableData = get(this, 'tableData');
+    const includedOptions = new Set();
+
+    return tableData.reduce(
+      (output, row) => {
+        const { category, severity } = row;
+
+        !includedOptions.has(category) && output.category.push({ label: category, value: category });
+        severity && !includedOptions.has(severity) && output.severity.push({ label: severity, value: severity });
+        // Ensures we only add each option once, assumes severity will never === category
+        includedOptions.add(category);
+        includedOptions.add(severity);
+
+        return output;
+      },
+      {
+        category: <IDropDownOption<string>[]>[{ label: 'Category', value: '' }],
+        severity: <IDropDownOption<string>[]>[{ label: 'Severity', value: '' }]
+      }
+    );
+  });
+
+  /**
+   * Passed in table data, mostly raw detailed information about the compliance score details and the
+   * breakdown of such.
+   * @type {Array<IHealthScore>}
+   */
+  tableData: Array<IHealthScore>;
+
+  /**
+   * Passed in severity filter. This property lives on the dataset-health container component and its
+   * modification is triggered by a click in the graphing component
+   * @type {string}
+   */
+  currentSeverityFilter: string;
+
+  /**
+   * Passed in category filter. This property lives on the dataset-health container component and its
+   * modification is triggered by a click in the graphing component
+   * @type {string}
+   */
+  currentCategoryFilter: string;
+
+  /**
+   * Calculates table data from the passed in information by appending each row with information that helps to
+   * style the table.
+   * @type {ComputedProperty<Array<IRenderedHealthScore>>}
+   */
+  renderedTableData: ComputedProperty<Array<IRenderedHealthScore>> = computed(
+    'tableData',
+    'currentCategoryFilter',
+    'currentSeverityFilter',
+    function(this: DatasetsHealthScoreTable): Array<IRenderedHealthScore> {
+      const { tableData, currentCategoryFilter: categoryFilter, currentSeverityFilter: severityFilter } = getProperties(
+        this,
+        'tableData',
+        'currentCategoryFilter',
+        'currentSeverityFilter'
+      );
+      return tableData.map(healthScore => ({
+        ...healthScore,
+        isHidden:
+          (!!categoryFilter && healthScore.category !== categoryFilter) ||
```

Comment:
Can we simplify this operation, it seems more complex than it needs to be and a lot of cognitive load to decipher what's going on :(